terraform/shared/shared.tf.json.template.py

```diff
@@ -7,17 +7,23 @@
     emit_tf,
 )
 
+# The provider of the S3 bucket used by each deployment stage.
+bucket_providers = {
```

Comment:
In light of the upcoming AnVIL deployments, I'm afraid we need to make these things (bucket providers and trail names) configurable. These will need to be lower-case named env variables containing JSON.

The provider override should be as light-footed as possible. In a completely new account, the default provider should be used so that the region is controlled by AWS_DEFAULT_REGION. A provider override should only occur on a resource if the resource is 1) imported and 2) resides in a region different from the one configured via `AWS_DEFAULT_REGION`.